### ISSUE TITLE
Apple Pay Token

### ISSUE THREAD

**Issue by achall9** [stripe/stripe-react-native#915]
**Is your feature request related to a problem? Please describe.**
It would be great to create a one-time use apple pay token so I can authorize a charge and capture the payment at a future time once.

**Describe the solution you'd like**
I would like the presentApplePay sheet to return a token as well as paymentMethod and Error.

**Describe alternatives you've considered**
The alternative is using the intent API but it is not possible currently for us.

**Comment by ashar-sarwar-viabletree** [comment]
any update on this ?

**Comment by charliecruzan-stripe** [comment]
Just to clarify- [this is the token you are referring to](https://developer.apple.com/documentation/passkit/pkpayment/1619239-token), correct?

**Comment by russmenum** [comment]
Any news here trying to migrate from ```tipsi-stripe``` and the lack of Native Pay tokens like we had there may be a problem.

The migration guide assume no one was using
```
const applePaytoken = await Stripe.paymentRequestWithApplePay(items, optionsApplePayToken)
```
or 

```
const androidPaytoken = await Stripe.paymentRequestWithAndroidPay(optionsAndroidPay)
```

Based on the doc I am not following how to send a 
```
body = {
        ...body,
        NativePayToken: returnedNativePay?.tokenId,
      }
```

Like you got from ```applePaytoken``` or ```androidPaytoken``` in the above example like you did from ```tipsi-stripe```


None of the "after" links in the [migration guide](https://github.com/stripe/stripe-react-native/blob/master/docs/tipsi-stripe-migration-guide.md) work so some clarity is missing.

So there is no clerity if ```const { error, paymentMethod } = await presentApplePay()``` and or ```const { error: presentError } = await presentGooglePay()``` return this token we had with  ```tipsi-stripe```?

**Comment by charliecruzan-stripe** [comment]
This work is in progress here: https://github.com/stripe/stripe-react-native/pull/1164 :)

**Comment by russmenum** [comment]
> This work is in progress here: #1164 :)

I saw that linked to at the top, and that it is failing review at this time...  So is this ```tipsi-stripe``` supported feature not available at this time in  ```stripe-react-native```?

And assuming it becomes available? Would it only be in the latest ver? I ask because of the 
```
compileSdkVersion = ??
targetSdkVersion = ??
```
ongoing issue where not all versions of this library are available to an app because of the need to target the current minimum standard rather than necessarily the latest targets and compile on Android? Particularly with respect to ```expo install @stripe/stripe-react-native``` limitations? 

While I can understand WIP, many of us are basically under a forced migration from ```tipsi-stripe``` , so  a lack of features it had is an issue, because it breaks payment for our apps.


Presuming, I understand correctly that this is just not here right now: is there a snack or doc any where of a more complete migration to get the same net result to complete migration where this was used? With the links in the migration guide broken, I am guessing at some parts due to a lack of doc. Was able to find the missing 
```
const {
        isGooglePaySupported,
      } = useGooglePay();
```
That was actually not in the migration guide because the link to the docs where it was worked, but with the other links not working it is difficult to resolve how to work around these kinds of not 1:1

**Comment by charliecruzan-stripe** [comment]
Getting the Stripe token using Apple Pay and Google Pay is not supported right now, that's correct.

Once that feature is available, it would only be available on the version it's released on, and for each version thereafter

If you are using the expo managed workflow and not using `eas build`, then yes you are limited since you cannot change any of your native library versions at all. 

Question though- how are you using `tipsi-stripe` if you're using Expo? It is not available in the managed workflow.

**Comment by russmenum** [comment]
@charliecruzan-stripe do we have any idea on the time line for that pull request and getting the same native pay functionality TIPSI had to be part of ```@stripe/stripe-react-native``` our migration is largely done.

So the hold up for out iOS app is sole the lack of features TIPSI supported.

Android side is the whole API 31 Vs 32 but as Google Pay will need the tokens over there no point in working that until ```@stripe/stripe-react-native``` is ready to fully replace TIPSI

**Comment by russmenum** [comment]
What is the status of this WIP?

**Comment by SavageWilliam** [comment]
> This work is in progress here: #1164 :)

+1 :)

**Comment by charliecruzan-stripe** [comment]
This was released in v0.22.0

**Comment by SavageWilliam** [comment]
> This was released in v0.22.0

I've upgraded to 0.22.0, however `createPlatformPayToken` doesn't seem to be returned from `usePlatformPay`  as per the example?

**Comment by charliecruzan-stripe** [comment]
@SavageWilliam `createPlatformPayToken` isn't released just yet, that was added in as a fix. We'll be releasing that later this week 😄
